projects/core/src/lib/mask.ts

```diff
@@ -130,25 +132,35 @@ export class Maskito extends MaskHistory {
     }
 
     private handleDelete(event: Event, isForward: boolean): void {
+        const initialState: ElementState = {
+            value: this.elementState.value,
+            selection: extendToNotEmptyRange(this.elementState.selection, isForward),
+        };
+        const [initialFrom, initialTo] = initialState.selection;
         const {elementState} = this.options.preprocessor({
-            elementState: this.elementState,
+            elementState: initialState,
+            data: '',
+            actionType: isForward ? 'deleteForward' : 'deleteBackward',
         });
-        const [from, to] = extendToNotEmptyRange(elementState.selection, isForward);
         const maskModel = new MaskModel(elementState, this.options);
+        const [from, to] = elementState.selection;
 
         maskModel.deleteCharacters([from, to]);
 
         const newElementState = this.options.postprocessor(maskModel);
         const newPossibleValue =
-            elementState.value.slice(0, from) + elementState.value.slice(to);
+            initialState.value.slice(0, initialFrom) +
+            initialState.value.slice(initialTo);
 
         if (newPossibleValue === newElementState.value) {
             return;
         }
 
         event.preventDefault();
 
-        if (areElementValuesEqual(elementState, maskModel, newElementState)) {
+        if (
+            areElementValuesEqual(initialState, elementState, maskModel, newElementState)
```

Comment:
what about
`areElementValuesEqual({ initialState, elementState, maskModel, newElementState })`

Reply:
Order of arguments doesn't matter for this function)

https://github.com/Tinkoff/maskito/blob/1855c3587b70522b0d1c196d1fe036e8bc11ae71/projects/core/src/lib/utils/element-states-equality.ts#L3-L6